mlflow/environment.py

```diff
@@ -0,0 +1,37 @@
+import yaml
+
+from mlflow.utils import PYTHON_VERSION
+from mlflow.version import VERSION as MLFLOW_VERSION
+
+
+class CondaEnvironment(object):
```

Comment:
Is this meant to be a public class/module or just an internal one?

Reply:
I meant for this to be public. Environments come up all the time and I thought it would be nice to have a convenience wrapper around it.

Reply:
I don't think we should make it a class for now, mostly because there are other things in Conda that we don't capture here yet, and it creates a maintenance burden for us. Can you switch the Spark functions to take the Conda environment in a file like we do in pyfunc for now? It's also inconsistent right now because Spark takes a CondaEnvironment but our other functions take a file.

Reply:
(And make this class private if we still want to use it, or just hold onto it to have another PR to add it later.)